eclass/java-pkg-simple.eclass

```diff
@@ -453,18 +565,42 @@ java-pkg-simple_src_compile() {
 			|| die "updating MANIFEST.MF failed"
 		rm -f "${T}/add-to-MANIFEST.MF" || die "cannot remove"
 	fi
+	JAVA_INTERNAL_JAR_FILES+=( "${PWD}/${JAVA_JAR_FILENAME}" )
+}
+
+# @FUNCTION: java-pkg-simple_src_compile
+# @DESCRIPTION:
+# src_compile for simple bare source java packages. Compiles a single
+# jar via java-pkg-simple_compile_jar and calls
+# java-pkg-simple_call_ejavadoc to generate documentation if USE FLAG
+# 'doc' exists and is set.
+java-pkg-simple_src_compile() {
+	java-pkg-simple_compile_jar
+
+	# javadoc
+	if has doc ${JAVA_PKG_IUSE} && use doc; then
+		if has binary ${JAVA_PKG_IUSE} && use binary; then
+			ewarn "Generation of documentation is not supported when installing binary."
+		elif [[ ${JAVADOC_SRC_DIRS} ]]; then
+			einfo "JAVADOC_SRC_DIRS exists, you need to call java-pkg-simple_call_ejavadoc separately"
```

Comment:
Haven't looked at the logic, but this appears to be a comment towards the ebuild developer and not the Gentoo users. In this case, this should maybe a `die` instead of `einfo`?  Or at least a `eqawarn`.

Reply:
Before my changes, this was just an information for the developer, that was **always** shown when `JAVADOC_SRC_DIRS` was used. A `die` would have prevented a use of that feature all together. But after my changes one could prevent this notice by calling `java-pkg-simple_compile_jar` instead of `java-pkg-simple_src_compile`. So maybe we should change this indeed to an `eqawarn` and suggest to not use `java-pkg-simple_src_compile` together with `JAVADOC_SRC_DIRS`.

(A `die` would break existing, functioning ebuilds. I would not do this at this point. Maybe after some time with `eqawarn`)

Reply:
> Before my changes, this was just an information for the developer, that was **always** shown when `JAVADOC_SRC_DIRS` was used. A `die` would have prevented a use of that feature all together. But after my changes one could prevent this notice by calling `java-pkg-simple_compile_jar` instead of `java-pkg-simple_src_compile`. So maybe we should change this indeed to an `eqawarn` and suggest to not use `java-pkg-simple_src_compile` together with `JAVADOC_SRC_DIRS`.
> 
> (A `die` would break existing, functioning ebuilds. I would not do this at this point. Maybe after some time with `eqawarn`)

if @fordfrog agrees, [that message](https://github.com/gentoo/gentoo/pull/34108#pullrequestreview-1798750973) could be dropped i guess.